test/emailpassword/passwordreset.test.js

```diff
@@ -754,4 +759,163 @@ describe(`passwordreset: ${printPath("[test/emailpassword/passwordreset.test.js]
+
+        app.use(middleware());
+
+        app.use(errorHandler());
+
+        user = await EmailPassword.signUp("public", "test@example.com", "password1234");
+        link = await createResetPasswordLink("public", user.user.id, "test@example.com");
+        assert(link !== undefined);
+        assert(link.status === "OK");
+
+        parsed = url.parse(link.link, true);
+
+        assert(parsed.pathname === "/auth/reset-password");
+        assert(parsed.query.token !== undefined);
+        assert(parsed.query.rid === "emailpassword");
+        assert(parsed.query.tenantId === "public");
+    });
+
+    it("test the reset password link for invalid input", async function () {
+        const connectionURI = await startST();
+        let emailPasswordLink = "";
+        STExpress.init({
+            supertokens: {
+                connectionURI,
+            },
+            appInfo: {
+                apiDomain: "api.supertokens.io",
+                appName: "SuperTokens",
+                origin: ({ request }) => {
+                    return "localhost:3000";
+                },
+            },
+            recipeList: [
+                EmailPassword.init({
+                    emailDelivery: {
+                        override: (original) => {
+                            return {
+                                ...original,
+                                sendEmail: async (input) => {
+                                    emailPasswordLink = input.passwordResetLink;
+                                },
+                            };
+                        },
+                    },
+                }),
+            ],
+        });
+
+        let link = await createResetPasswordLink("public", "invlidUserId", "test@example.com");
+        assert(link !== undefined);
+        assert(link.status === "UNKNOWN_USER_ID_ERROR");
+
+        try {
+            link = await createResetPasswordLink("invalidTenantId", "invlidUserId", "test@example.com");
+        } catch (err) {
+            isErr = true;
+            assert(err.message.includes("status code: 400"));
+        }
+        assert(isErr);
+    });
+
+    it("test sendResetPasswordEmail", async function () {
+        const connectionURI = await startST();
+        let emailPasswordLink = "";
+        STExpress.init({
+            supertokens: {
+                connectionURI,
+            },
+            appInfo: {
+                apiDomain: "api.supertokens.io",
+                appName: "SuperTokens",
+                origin: ({ request }) => {
+                    return "localhost:3000";
+                },
+            },
+            recipeList: [EmailPassword.init()],
+        });
+
+        const app = express();
+
+        app.use(middleware());
+
+        app.use(errorHandler());
+
+        user = await EmailPassword.signUp("public", "test@example.com", "password1234");
+        resp = await sendResetPasswordEmail("public", user.user.id, "test@example.com");
```

Comment:
you should override the sendEmail function and check that the link being sent is actually correct.